components/template-mgt/org.wso2.carbon.identity.template.mgt/src/main/java/org/wso2/carbon/identity/template/mgt/internal/TemplateManagerComponent.java

```diff
@@ -100,4 +129,119 @@ protected void unsetConfigurationManager(ConfigurationManager configurationManag
         }
         TemplateManagerDataHolder.getInstance().setConfigurationManager(null);
     }
+
+    @Reference(
+            name = "identity.template.handler",
+            service = ReadOnlyTemplateHandler.class,
+            cardinality = ReferenceCardinality.OPTIONAL,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetReadOnlyTemplateHandler")
+    protected void setReadOnlyTemplateHandler(ReadOnlyTemplateHandler readOnlyTemplateHandler) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Template handler " + readOnlyTemplateHandler.getClass().getName() + " is added.");
+        }
+        TemplateManagerDataHolder.getInstance().addReadOnlyTemplateHandler(readOnlyTemplateHandler);
+    }
+
+    protected void unsetReadOnlyTemplateHandler(ReadOnlyTemplateHandler readOnlyTemplateHandler) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Template handler " + readOnlyTemplateHandler.getClass().getName() + " is removed.");
+        }
+        TemplateManagerDataHolder.getInstance().removeReadOnlyTemplateHandler(readOnlyTemplateHandler);
+    }
+
+    private void loadDefaultTemplates() throws FileNotFoundException {
+
+        List<Path> paths = Arrays.asList(TemplateMgtConstants.SP_TEMPLATES_DIR_PATH, TemplateMgtConstants
+                .IDP_TEMPLATES_DIR_PATH);
+        for (Path path : paths) {
+            if (!Files.exists(path) || !Files.isDirectory(path)) {
+                log.warn("Templates directory not found at " + path);
+                throw new FileNotFoundException("Templates directory not found at " + path);
+            } else {
+                try {
+                    Files.walk(path)
+                            .filter(filePath -> Files.isRegularFile(filePath) && filePath.toString().endsWith
+                                    (TemplateMgtConstants
+                                            .FILE_EXT_JSON))
+                            .forEach(filePath -> {
+                                try {
+                                    String templateJsonString = FileUtils.readFileToString(filePath.toFile());
+                                    JSONObject templateObj = new JSONObject(templateJsonString);
+                                    Template template = new Template();
+
+                                    template.setTemplateId(templateObj.getString(TemplateMgtConstants.ID));
+                                    template.setTemplateName(templateObj.getString(TemplateMgtConstants.NAME));
+                                    template.setDescription(templateObj.getString(TemplateMgtConstants.DESCRIPTION));
+                                    template.setImageUrl(templateObj.getString(TemplateMgtConstants.IMAGE));
+
+                                    if (StringUtils.equals(path.toString(), TemplateMgtConstants
```

Comment:
as a practice, please use the constant attribute as the first element. Nothing wrong with the code :) as you are using StringUtils.equals()

Reply:
addressed in 26d72337ba0abdf9a51c5a47dcef5591353767af